packages/modules/viewer-react/src/types.ts

```diff
@@ -134,20 +141,59 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
 
 export interface ConnectedViewerProps {
   iTwinId: string;
-  iModelId: string;
+  iModelId?: string;
   changeSetId?: string;
 }
-
 export interface FileViewerProps {
   /** Path to local snapshot or briefcase */
   filePath: string;
 }
 
-export interface BlankViewerProps {
-  blankConnection: BlankConnectionProps;
+export type BlankViewerProps = {
+  /** @deprecated specify location and extents instead. */
+  blankConnection?: BlankConnectionProps;
+  location?: Cartographic | EcefLocationProps;
+  extents?: Range3dProps;
+  blankConnectionViewState?: BlankConnectionViewState;
+};
+
+type BlankViewerPropsWithoutDeprecation = {
+  blankConnection?: BlankConnectionProps;
+  location?: Cartographic | EcefLocationProps;
+  extents?: Range3dProps;
   blankConnectionViewState?: BlankConnectionViewState;
+};
+
+export interface BlankConnectionInitializationProps {
+  iTwinId?: string;
+  blankConnectionProps: BlankConnectionProps;
+}
+
+export interface RequiredViewerConnectionProps {
+  iTwinId?: string;
+  iModelId?: string;
+  filePath?: string;
+  blankConnection?: BlankConnectionProps;
+  extents?: Range3dProps;
+  location?: Cartographic | EcefLocationProps;
 }
 
+export enum ConnectionType {
+  None,
```

Comment:
this really shouldnt be a valid case, and i think the way the types were setup if you didnt satisfy the XOR union it would yell at you that youre missing props

Reply:
Removed ConnectionType. Was adding it to aid in discriminated unions, but using your `XOR` was fine.